### PR TITLE
[Refactor][UT] Refactor and add unit tests for SinkIOBuffer

### PR DESCRIPTION
**Pull request by letian-jiang**
Moves common logic to the `SinkIOBuffer` class.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [x] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exec/pipeline/sink/sink_io_buffer.h [hunk 1]
```diff
@@ -105,33 +124,64 @@ class SinkIOBuffer {
     }
 
     static int execute_io_task(void* meta, bthread::TaskIterator<ChunkPtr>& iter) {
+        auto* sink_io_buffer = static_cast<SinkIOBuffer*>(meta);
```
Comment (zombee0) [inline be/src/exec/pipeline/sink/sink_io_buffer.h:127]: I am not sure, the last sinker close the execution queue, but weather it's possible that there are more than one task in queue when closing?

File: be/src/exec/pipeline/sink/sink_io_buffer.h [hunk 2]
```diff
@@ -51,45 +48,67 @@ class SinkIOExecutor : public bthread::Executor {
...
+        CHECK(false); // unreachable
     }
 
-    virtual bool is_finished() { return _is_finished && _num_pending_chunks == 0; }
+    bool is_finished() { return _is_closed && _num_pending_chunks == 0; }
 
-    virtual void cancel_one_sinker() { _is_cancelled = true; }
+    void cancel_one_sinker() { _is_cancelled = true; }
 
     virtual void close(RuntimeState* state) {
```
[the remark this answers is not in the document]
Comment (letian-jiang) [reply be/src/exec/pipeline/sink/sink_io_buffer.h:103]: Here being canceled is a state before closed, signaling subsequent IO tasks can be skipped.

File: be/src/exec/pipeline/sink/sink_io_buffer.h (hunk 1 above)
Comment (letian-jiang) [reply be/src/exec/pipeline/sink/sink_io_buffer.h:127]: Nope. Stop task is the very last task in queue.

**Review by zombee0 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 31 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/file_sink_operator.cpp|0|4|00.00%|[67, 103, 105, 109]|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|5|00.00%|[36, 62, 77, 78, 118]|
|:large_blue_circle:|src/exec/pipeline/sink/mysql_table_sink_operator.cpp|0|4|00.00%|[34, 68, 70, 73]|
|:large_blue_circle:|src/exec/pipeline/sink/sink_io_buffer.h|0|18|00.00%|[56, 60, 61, 62, 64, 67, 68, 81, 84, 86, 88, 89, 90, 95, 96, 104, 108, 111]|

**Pull request closed without merging**